modules/fedimint-ln/src/lib.rs

```diff
@@ -939,6 +1000,25 @@ impl LightningModule {
             .await
             .expect("DB error");
     }
+
+    async fn process_clock_time_proposals(&self, mut proposals: Vec<SystemTime>) -> SystemTime {
+        assert!(!proposals.is_empty());
+
+        proposals.sort();
+        let median_proposal = match proposals.len() % 2 {
+            0 => {
+                let earlier_time = proposals[proposals.len() / 2 - 1];
+                let later_time = proposals[proposals.len() / 2];
+                let time_diff = later_time.duration_since(earlier_time);
+                earlier_time
+                    + time_diff
+                        .expect("Error calculating median time")
+                        .div_f32(2.0)
+            }
+            _ => proposals[proposals.len() / 2],
+        };
```

Comment:
That might be a bit over the top, I'd rather avoid the floating point math. As long as the chosen element is between the elements of two honest parties it's ok (e.g. with elements 1,2,3,4 and up to one malicious party both 2 and 3 are ok to choose). Just choose one.

Reply:
I simplified the clock time proposal in https://github.com/fedimint/fedimint/pull/1074/commits/24a5d5e2595e51a17b2d21a910cd0361e5834f08